Makefile.Common

```diff
@@ -0,0 +1,140 @@
+# SRC_ROOT is the top of the source tree.
+SRC_ROOT := $(realpath $(dir $(lastword $(MAKEFILE_LIST))))
```

Comment:
I think establishing this in a separate PR may be better than associating it w/ these changes directly since they just need to include it.

Reply:
Good idea, opening #2040